food-waste-ui/src/components/App/App.jsx

```diff
@@ -119,34 +119,43 @@ export function MainApp({
           setCurrentUser(curUser);
 
           let allProducts = response.data.products;
-          //TODO: filter products for search
-          setProducts(allProducts);
+          let newProducts = allProducts.filter((item) =>
+            item.name.toLowerCase().includes(searchTerm.toLowerCase())
```

Comment:
Something to think about is user errors whenever theres a form/input. What happens if there is extra trailing/leading whitespace or if the user spells the searchTerm incorrectly? A stretch could be to add suggested search results based on the user input with errors.